### ISSUE TITLE
Huwawei Appgallery purchase plugin

### ISSUE THREAD

**Issue by rafaellop** [j3k0/cordova-plugin-purchase#1093]
Huawei made available their plugin for inapp purchases from cordova apps for use in the Huawei Appgallery store. It would be great to have that store supported in the @j3k0  `cordova-plugin-purchase` plugin. What do you think?

**Comment by j3k0** [comment]
Hey, if the license is compatible we ca integrate the code and write a "bridge". I'll contact them.

**Comment by j3k0** [comment]
I'm a bit lost on their website, did you find where the we can download the cordova plugin ?

**Comment by rafaellop** [comment]
@j3k0 I've just emailed you.

**Comment by j3k0** [comment]
I didn't get an email. Did you use hoelt@fovea.cc ?

**Comment by rafaellop** [comment]
Yes, I've used the email. It seems you may have my ISP domain blocked or something. I can try to send it from a different domain. Please let me know.

**Comment by NLueg** [comment]
What is the status of supporting the Huawei AppGallery? I would really appreciate it.

**Comment by rafaellop** [comment]
I can send it to @j3k0  but it seems my email is bouncing.

**Comment by j3k0** [comment]
Hey @rafaellop - Yes, I didn't get the email (maybe the file is too big for GMail?)

I tried again browsing around Huawei website, to google "cordova huawei plugin", ... no luck. All I could do is reaching some "coming soon" pages. Where did you find the cordova plugin?

Thanks

**Comment by rafaellop** [comment]
[cordova-plugin-hms-inapppurchases-4.0.3.zip](https://github.com/j3k0/cordova-plugin-purchase/files/5239385/cordova-plugin-hms-inapppurchases-4.0.3.zip)

Please delete when downloaded.

It's available for developers on Huawei website.

**Comment by j3k0** [comment]
Got it! Thanks @rafaellop

**Comment by j3k0** [comment]
Anyone has experience working with AppGallery? I think it would require a new plugin (this one with Google integration and another one for the huawei integration). I'm thinking about only providing a support javascript file that implements the same API as `cordova-plugin-purchase` over the HMS plugin provided by Huawei.

Also, apparently the sandbox mode for Huawei AppGallery development is not available outside of China... How convenient.

**Comment by rafaellop** [comment]
I don't have any experience with the AppGallery. They just contacted me encouraging to port my app to their ecosystem, but firstly they didn't support Cordova. Recently got contacted again and their sent me information about the plugin I told you about. I've sent a question to my consultant in Huawei about the sandbox and testing possibilities in general and waiting for a reply few days already.

**Comment by yoprogramo** [comment]
I have downloaded the code and it is licensed Apache v2.0 so we can provide you (I guess)... I'll try to send to you to see if it is possible to integrate this purchase method.

**Comment by simion** [comment]
I would need this too for https://clandesto.app
Got IAP working on Android/IOS, published app also on huawei AppGallery, but i would rather have one implementation for IAP.
